activiti-api-task-runtime/src/main/java/org/activiti/api/task/runtime/TaskAdminRuntime.java

```diff
@@ -74,5 +84,21 @@
 
     void setVariables(SetTaskVariablesPayload setTaskVariablesPayload);
 
-
+    /**
+     * Assign a task with a new user
+     *  - If the currently authenticated user is not a candidate throw an IllegalStateException
+     *  - If there is a user assigned, reassign it to the new user
+     *  - after the claim the task should be in assigned status
```

Comment:
@CTI777 if assign set an assignee, there is no need to claim. You only need to claim when there is no assignee and only candidate users or groups. Does that make sense?